Newsletters: add #22 (2018-11-20)

harding opened the pull request:
Our longest newsletter yet.  Hopefully that doesn't scare readers away.

If we can get a LN expert to review the feature news section, I'd love that.  In case we can, hopefully I've described things at high enough level that I didn't get too much wrong.

practicalswift commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,319 @@
+necessarily defined or agreed upon yet---but the basic outline of the
+new features is available.  In linking to the [working outline][ln1.1
+outline], Rusty Russell noted the following highlights: "multi-path
+payments, dual-funded channels, splicing , wumbo, hidden destinations,
```
Nit: Remove space after splicing :-)

jnewbery commented:
@cdecker @renepickhardt - if you have time to review the LN sections, that'd be amazing. We publish Tuesday morning (US east coast time).

cdecker commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,319 @@
+  the same parties.  Closing one channel and opening another requires
+  completely stopping all payments between the two parties until an
+  appropriate number of onchain confirmations have been received for the
+  close-and-reopen transaction.
```
Even more importantly splice-outs allow you to perform on-chain payments directly out of an off-chain contract. This means that there is no need to differentiate between off-chain and on-chain funds anymore. This goes a long way towards a much simpler UX where users don't have to differentiate, and in combination with AMP it allows us to hide the channel management completely from the user.

cdecker commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,319 @@
+
+    For more information, see the following threads where this feature
+    is also called rendez-vous routing: [1][cjp rz], [2][zmn rz], [3][zmn
+    rz packetswitch].
```
For a detailed writeup you can point to https://github.com/lightningnetwork/lightning-rfc/wiki/Rendez-vous-mechanism-on-top-of-Sphinx. We're trying to aggregate the discussions in the Wiki so it should get easier to get a complete view of the proposal instead of having to read through the evolution in the form of mails :wink:

cdecker replied:
Also if anybody has an idea on how to write that up in a more concise and precise way I'd be happy to hear :-)

harding commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,319 @@
+    incentive mechanism that can reward capital providers like Bob is
+    still being discussed.  For more information, see the following
+    threads: [1][neigut liquidity], [2][zmn liquidity], [3][zmn dual
+    rbf].  See also [Newsletter #21][].
```
I'm adding a note for "the initiator of the channel currently also pays the fees for closing the channel" as you suggest, but I don't see any other easy changes---it seems to me, as is said in the post, that finding the best incentive mechanism for dual-funded is still being discussed.  (One of the nice things about having a weekly newsletter, is that we often get the opportunity to write follow-up stories---so there's usually no need to go too deep on things that are still being worked on.  For example, this particular week's newsletter adds detail to items mentioned in three previous weeks.)

Is the symmetric CSV comment directly regarding dual-funded or is it a suggestion for a separate item?  I see https://github.com/lightningnetwork/lightning-rfc/wiki/Brainstorming#symmetric-csv-delay which does sound interesting and I see that it's one of the accepted proposals in the 1.1 spec draft.  I didn't mention CSV here since it wasn't in Rusty's highlights I used as a filter, but I'm happy to write it up if you think it's noteworthy (I just need to know whether it should go with dual-funded or be a separate item).

moneyball commented:
I reviewed the non-LN parts. The section on fees and hash rate looks good to me. I'll try to review the LN section from a lay person perspective after any changes based on LN experts review.

harding commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,319 @@
+
+    For more information, see the following threads where this feature
+    is also called rendez-vous routing: [1][cjp rz], [2][zmn rz], [3][zmn
+    rz packetswitch].
```
Oh, that is very important.  Thank you!  I didn't read past the first post in that thread before linking to it (sorry, you people put in a crazy amount of work last week and I'm still trying to catch up!).  I just did my homework and I think even more important is Decker's extended remark on Zmn's base AMP proposal at https://lists.linuxfoundation.org/pipermail/lightning-dev/2018-November/001573.html, Fromknecht's reply, and the rest of that sub-thread.

Since all of those are encompassed by the linked threads, what I'm going to do is keep those links the same but provide a prominent link to https://github.com/lightningnetwork/lightning-rfc/wiki/Rendez-vous-mechanism-on-top-of-Sphinx as suggested by Decker subsequent to your comment above.  Of course, please let me know if you disagree or see a better option.

harding commented:
I pushed a bunch of commits (eventually to be squashed) with one commit per comment or suggestion, labeling each commit with the name of the person who made the suggestion.  I also also left replies on a couple of the comments.  I think that should address all feedback.

I also added a Special Thanks section thanking y'all for your feedback, as I'm very grateful to have it.

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+Specification 1.1.  Thirty proposals were accepted at a
+high-level---meaning full specifications for each proposal are not
+necessarily defined or agreed upon yet---but the basic outline of the
+new features is available.  Below, Optech has attempted to briefly
```
Actually, also remove 'many gossip improvements', since you've removed the section on that.

harding replied:
Dropping the whole quote and moving the link earlier in the paragraph.  I was trying to be clever by letting Russell choose the topics, but it was quite wasteful to try to demonstrate my cleverness---especially after we removed the gossip point and added the not-spec'd paragraph.

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+  routing path including Bob until Alice has sent Bob some money.  This
+  creates a bootstrapping problem: if Alice wants to receive payments
+  via LN, she has to get people to open new channels to her node---which
+  requires them paying possibly-high onchain fees and waiting for onchain
```
"requires them paying .. and waiting" or "requires them to pay ... and wait"

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+
+The reason long interblock periods become more common at a faster rate
+than average interblock times increase is because the variance for
+independent events is the square of the average.
```
Great commenting! :kissing: :ok_hand:

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+      preparing any fee-reduction measures you're willing to use such as
+      payment batching.
+
+    - *Increased revenues for profit-maximizing miners:* each time
```
Note that mining becomes more profitable even before a difficulty adjustment because fees become a more significant proportion of coinbase reward (eg https://blockstream.info/tx/09ed9d39d5dab626b8d55dd0b36e63eaf15d7e19b34fa8effaae2cba8381d43c had almost 0.9BTC of fees)

harding replied:
I think your comment is just saying that higher fees (described in the text's previous bullet point) mean more miner profit, ceteris paribus.  That's an obvious insight I missed, so I'm adding it.  However, the way you've phrased it makes me wonder whether I'm missing some subtler point---if so, please let me know!

jnewbery replied:
no, that's it exactly. Miners get higher fees until the downwards difficulty adjustment, and then they get more frequent blocks. Christmas has come early!

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+  10% of the list's traffic in the past 365 days.  Many of the threads
+  continue conversations started at the protocol developers summit.  If
+  you're interested in Lightning protocol development, we suggest
+  reading each of this month's [threads][ln list].
```
Was this supposed to link to a thread view of the LN mailing list? It links to the list info page.

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+
+## Special thanks
+
+We thank Chris Decker, PracticalSwift, and Rene Pickhardt for providing
```
Christian Decker, practicalswift, and René Pickhardt

harding replied:
D'oh.  Thanks!

jnewbery commented:
I've reviewed e9bb477ada66b810325ee688d9cbe89f4e45e106 and confirmed that all fixup commits do what they say. I'm happy for them to be squashed (not sure if Dave wants to wait until the other reviewers have had a chance to review the individual fixup commits).

jnewbery commented:
oh, also added an "Bitcoin Optech News" section with a note that the second workshop was held.

harding commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+  routing path including Bob until Alice has sent Bob some money.  This
+  creates a bootstrapping problem: if Alice wants to receive payments
+  via LN, she has to get people to open new channels to her node---which
+  requires them paying possibly-high onchain fees and waiting for onchain
```
This particular newsletter issue is completely consistent with unhyphenated---I made sure of that.  Alas, a quick git grep shows we've been completely inconsistent in the past.  I went with unhyphenated here as it gets to be quite a few hyphens when talking about things like splicing, but I also don't care much.

Putting together a style guide has been on my mental todo list since we started.  I should probably do that at some point.

harding commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,341 @@
+    receives multiple payments within a reasonable time period that
+    equal or exceed the expected amount, he can reveal their shared
+    preimage to claim them simultaneously---providing the same security
+    currently used for single-path payments.  This also works if some
```
A new paragraph!  And ruin my precious parallelism between each of these points!  Are you crazy!  :-)

Seriously, I'm confused by your remark.  Splitting further along the path was something I deliberately didn't describe because the [base AMP proposal](https://lists.linuxfoundation.org/pipermail/lightning-dev/2018-November/001577.html) indicated it might be unsafe so I don't want to promise it.  What's described here is merging along the path---to make that clearer, I'm revising the phrasing slightly.

I don't think a single sentence about path merging is enough to warrant a new paragraph, so I'm leaving the sentence where it is (feel free to overrule), but if you want an upgrade to an additional sentence about non-spender path splitting, I'll throw in the extra paragraph for free.  :-)

jnewbery replied:
I think it looks great as it is in your latest commit.

I hadn't read the Base AMP emails before my comment and was going off my (limited) knowledge of OG AMP where I thought arbitrary splits were allowed and payment was guaranteed to be atomic.

I don't understand the point in zmn's email about the safety of arbitrary splits. Hopefully at some point there'll be an explanation on the mailing list of why that is.

harding commented:
I believed I've addressed @jnewbery's excellent suggestions (with some notes left in individual comments), done a new proofread with a few minor edits, and squashed down to one commit.  The original branch and the commit addressing John's points is here: https://github.com/harding/bitcoinops.github.io/commits/2018-11-20-newsletter-orig

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,351 @@
+
+- **Wumbo:** by agreement among early LN implementations, currently each
+  channel's capacity is limited by default to about 0.168 BTC (about $40
+  USD when defined; currently about $850).  This was [chosen][russell
```
sir when $1000?

jnewbery commented on _posts/en/newsletters/2018-11-20-newsletter.md:
```diff
@@ -0,0 +1,351 @@
+    A proposed solution to this problem is to allow dual-funded
+    channels.  Alice agrees to put 0.1 BTC into a channel with Bob if
+    Bob agrees to open the channel with 0.1 BTC of his own funds.  This
+    can cost Bob money---namely onchain transaction fees and lost
```
nit: is 'lost opportunity cost' right? Should it be 'onchain transaction fees and the opportunity cost'

jnewbery commented:
One additional small nit. Otherwise tACK da9dc36a9c9f6188927567b679af26a1d983e36d.

The newsletter is always great, but I think this may be the best yet. There's so much great information in here! Thanks Dave, and thanks to @cdecker @renepickhardt and @practicalswift for contributing.

harding commented:
Nit addressed and price-dependent note updated.  Thanks again everyone!

```diff
--- a/_posts/en/newsletters/2018-11-20-newsletter.md
+++ b/_posts/en/newsletters/2018-11-20-newsletter.md
@@ -77,8 +77,8 @@ new features is available.  Some highlights from the meeting include:
     A proposed solution to this problem is to allow dual-funded
     channels.  Alice agrees to put 0.1 BTC into a channel with Bob if
     Bob agrees to open the channel with 0.1 BTC of his own funds.  This
-    can cost Bob money---namely onchain transaction fees and lost
-    opportunity cost from having his funds committed for some time---but
+    can cost Bob money---namely onchain transaction fees and the
+    opportunity cost of having his funds committed for some time---but
     Bob also receives the opportunity to earn LN routing fees for any
     payments sent to Alice.
 
@@ -120,7 +120,7 @@ new features is available.  Some highlights from the meeting include:
 
 - **Wumbo:** by agreement among early LN implementations, currently each
   channel's capacity is limited by default to about 0.168 BTC (about $40
-  USD when defined; currently about $850).  This was [chosen][russell
+  USD when defined; currently about $750).  This was [chosen][russell
   why limit] to help prevent users from putting too much money into
   unproven software.
```

harding commented:
@jnewbery oh, I just realized that the bullet about mining was sounding pretty odd with the recent price drop, so I added a sentence providing an alternative perspective.  Feel free to remove or edit that addition freely (or anything else, as always).
